subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/accessors/AccessorsClassPath.kt

```diff
@@ -67,62 +87,139 @@ fun projectAccessorsClassPath(project: Project, classPath: ClassPath): Accessors
 
 
 private
-fun buildAccessorsClassPathFor(project: Project, classPath: ClassPath) =
-    configuredProjectSchemaOf(project)?.let { projectSchema ->
+fun buildAccessorsClassPathFor(project: Project, classPath: ClassPath): AccessorsClassPath? {
+    val workExecutor: WorkExecutor<ExecutionRequestContext, CachingResult> = project.serviceOf()
+    val cacheKeyBuilder: CacheKeyBuilder = project.serviceOf()
+    val workspaceProvider: KotlinDslWorkspaceProvider = project.serviceOf()
+    return configuredProjectSchemaOf(project)?.let { projectSchema ->
         // TODO:accessors make cache key computation more efficient
-        cachedAccessorsClassPathFor(project, cacheKeyFor(projectSchema, classPath)) { srcDir, binDir ->
-            withAsynchronousIO(project) {
-                buildAccessorsFor(
-                    projectSchema,
-                    classPath,
-                    srcDir = srcDir,
-                    binDir = binDir
-                )
-            }
+        val cacheKeySpec = cacheKeyFor(projectSchema, classPath)
+        val cacheKey = cacheKeyBuilder.build(cacheKeySpec)
+        workspaceProvider.withWorkspace(cacheKey) { workspace, executionHistoryStore ->
+            val sourcesOutputDir = File(workspace, "sources")
+            val classesOutputDir = File(workspace, "classes")
+            val work = GenerateProjectAccessors(
+                project,
+                projectSchema,
+                classPath,
+                cacheKey,
+                sourcesOutputDir,
+                classesOutputDir,
+                executionHistoryStore,
+                project.serviceOf(),
```

Comment:
I wonder if it would be better to add the type parameters for these method calls. Without an IDE, this is very hard to  decipher.

Reply:
Yeah, I share the same concern. I think it's a more general problem with how `Project.serviceOf()` is declared. The plan here is that we'll refactor this so that the services are properly injected instead of looked up every time using `serviceOf()`.

Reply:
We don't have the unqualified `serviceOf()` calls anymore.